src/map/WorldMap.kt

```diff
@@ -20,23 +25,26 @@ class CoordinateNodeWorldMap(private val nodes: MutableMap<Coordinate, Node>) :
                 }
             }
          }
-        throw NoSuchElementException("The provided actor is not in the map.")
+        throw NoSuchElementException(ACTOR_NOT_ON_MAP_EXCEPTION_MESSAGE)
     }
 
     override fun moveObject(actor: Actor, deltax: Int, deltay: Int) {
+        actor.loseOneLifeUnit()
```

Comment:
you probably want a player to lose life points every turn, rather than every move (ie he loses hp regardless of if he wants to move)
also, the entity to be moved parameter should probably be a MapObject, rather than an Actor

Reply:
yes agreed